pkg/kubernetes/delete.go

```diff
@@ -3,11 +3,20 @@ package kubernetes
 import (
 	"github.com/grafana/tanka/pkg/kubernetes/client"
 	"github.com/grafana/tanka/pkg/kubernetes/manifest"
+	"github.com/grafana/tanka/pkg/process"
 )
 
 type DeleteOpts client.DeleteOpts
 
 func (k *Kubernetes) Delete(state manifest.List, opts DeleteOpts) error {
+	// Sort and reverse the manifests to avoid cascading deletions
+	process.Sort(state)
+	for i := 0; i < len(state)/2; i++ {
+		t := state[i]
+		state[i] = state[len(state)-1-i]
+		state[len(state)-1-i] = t
+	}
```

Comment:
While this is recommended in SliceTricks, I just discovered there is `sort.Reverse` which can be used as long as the type implements `sort.Interface`.

We could consider whether it's worth moving most of the sort logic from `process` to `manifest`, and have `manifest.List` implement the interface, so that you could `sort.Sort(manifest.List{})` and also `sort.Reverse`

Reply:
I had a look at sort.Reverse and it looked like a lot of bureaucracy. If we migrated from sort.SliceStable, it might be a bit more worthwhile, but I'm not sure if it's that intuitive in this case; e.g. the sorting criterion is non-trivial. I'd give it a go in a separate PR, since this is an internal cleanup, not a user-visible feature.

Reply:
Yeah thought about doing that. The slice would then know how to sort itself, allowing us to use `sort.Stable` and `sort.Reverse` on it

Reply:
We can do that, but I agree it's minor internal cleanup